homeassistant/helpers/entityfilter.py

```diff
@@ -107,3 +107,64 @@ def entity_filter_4c(entity_id: str) -> bool:
         return entity_id in include_e
 
     return entity_filter_4c
+
+
+class FilterBuilder:
+    """Builder class for entity filters."""
+
+    def __init__(self, include_all: bool = False,
+                 include_domains: Optional[Iterable[str]] = None,
+                 include_entities: Optional[Iterable[str]] = None,
+                 exclude_domains: Optional[Iterable[str]] = None,
+                 exclude_entities: Optional[Iterable[str]] = None):
+        """Initialiser."""
+        self.include_all = include_all
+        self.include_domains = set(include_domains or [])
+        self.include_entities = set(include_entities or [])
+        self.exclude_domains = set(exclude_domains or [])
+        self.exclude_entities = set(exclude_entities or [])
+
+    def build(self) -> Callable[[str], bool]:
```

Comment:
I haven't looked through the PR, so don't know how this is used, but if it's called more than once, it should be cached.

Reply:
Yeh it's cached.